File: backend/test_api.py
#!/usr/bin/env python
"""
Quick test script to verify the backend API is working
"""
import requests
import json

def test_api():
    base_url = "http://localhost:5000"
    
    try:
        # Test health endpoint
        print("🔍 Testing API Health...")
        response = requests.get(f"{base_url}/api/health")
        print(f"✅ Health Check: {response.status_code}")
        print(f"📄 Response: {response.json()}")
        
        # Test user names endpoint
        print("\n🔍 Testing User Names...")
        response = requests.get(f"{base_url}/api/user_names")
        print(f"✅ User Names: {response.status_code}")
        data = response.json()
        print(f"📄 Found {len(data.get('user_names', []))} user names")
        
        # Test projects endpoint
        print("\n🔍 Testing Projects...")
        response = requests.get(f"{base_url}/api/projects")
        print(f"✅ Projects: {response.status_code}")
        data = response.json()
        print(f"📄 Found {len(data.get('projects', []))} projects")
        
        print("\n🎉 All API endpoints are working!")
        
    except requests.exceptions.ConnectionError:
        print("❌ Connection Error: Make sure the backend server is running on port 5000")
        print("   Run: python app.py (from the backend directory)")
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    test_api()

File: backend_node/test_api.js
const http = require('http');

const testEndpoints = [
    { path: '/api/health', method: 'GET' },
    { path: '/api/projects', method: 'GET' },
    { path: '/api/areas', method: 'GET' }
];

function testAPI() {
    console.log('Testing ArcSpatialDB Node.js API...\n');
    
    testEndpoints.forEach((endpoint, index) => {
        setTimeout(() => {
            const options = {
                hostname: 'localhost',
                port: 5000,
                path: endpoint.path,
                method: endpoint.method
            };

            const req = http.request(options, (res) => {
                let data = '';
                
                res.on('data', (chunk) => {
                    data += chunk;
                });
                
                res.on('end', () => {
                    console.log(`${endpoint.method} ${endpoint.path}`);
                    console.log(`Status: ${res.statusCode}`);
                    
                    try {
                        const json = JSON.parse(data);
                        if (endpoint.path === '/api/health') {
                            console.log(`Response: ${json.message}`);
                        } else if (endpoint.path === '/api/projects') {
                            console.log(`Projects found: ${json.projects ? json.projects.length : 0}`);
                        } else if (endpoint.path === '/api/areas') {
                            console.log(`Areas found: ${json.areas ? json.areas.length : 0}`);
                        }
                    } catch (e) {
                        console.log('Response:', data.substring(0, 100) + '...');
                    }
                    
                    console.log('---\n');
                });
            });

            req.on('error', (e) => {
                console.error(`Error testing ${endpoint.path}:`, e.message);
                console.log('---\n');
            });

            req.end();
        }, index * 1000);
    });
}

// Wait a bit for server to start, then test
setTimeout(testAPI, 2000);
